hackathon_site/dashboard/frontend/src/components/teamDetail/TeamPendingOrderTable/TeamPendingOrderTable.test.tsx

```diff
@@ -0,0 +1,102 @@
+
+describe("team pending order table", () => {
+    test("renders team pending order table", () => {
+        const { getByTestId } = render(<TeamPendingOrderTable />);
+        expect(screen.getByText("Requested Items")).toBeInTheDocument();
+
+        //loop through all pending orders
+        mockPendingOrdersInTable.forEach((currentOrder) => {
+            //loop through all the hardware in each order
+            currentOrder.hardwareInTableRow.forEach((currentRow) => {
+                // renders all hardware names
+                expect(
+                    within(
+                        getByTestId(`table-${currentOrder.id}-${currentRow.id}`)
+                    ).getByText(`${mockHardware[currentRow.id - 1].name}`)
+                ).toBeInTheDocument();
+
+                // renders all hardware model numbers
+                expect(
+                    within(
+                        getByTestId(`table-${currentOrder.id}-${currentRow.id}`)
+                    ).getByText(`${mockHardware[currentRow.id - 1].model_number}`)
+                ).toBeInTheDocument();
+
+                // renders all hardware manufacturers
+                expect(
+                    within(
+                        getByTestId(`table-${currentOrder.id}-${currentRow.id}`)
+                    ).getByText(`${mockHardware[currentRow.id - 1].manufacturer}`)
+                ).toBeInTheDocument();
+            });
+        });
+    });
+
+    test("All button changes the value correctly", () => {
+        const { getByTestId, getByRole } = render(<TeamPendingOrderTable />);
+
+        mockPendingOrdersInTable.forEach((currentOrder) => {
+            currentOrder.hardwareInTableRow.forEach((currentRow) => {
+                const allButton = within(
+                    getByTestId(`table-${currentOrder.id}-${currentRow.id}`)
+                ).getByTestId(`all-button`);
+
+                const select = within(
+                    getByTestId(`table-${currentOrder.id}-${currentRow.id}`)
+                ).getByRole(`button`);
+
+                //set the select option to 0
+                fireEvent.mouseDown(select);
+
+                const listbox = within(getByRole("listbox"));
+
+                fireEvent.click(listbox.getByText(`0`));
+
+                expect(
+                    within(
+                        getByTestId(`table-${currentOrder.id}-${currentRow.id}`)
+                    ).getByTestId(`select`)
+                ).toHaveTextContent("0");
+
+                fireEvent.click(allButton);
+
+                expect(
+                    within(
+                        getByTestId(`table-${currentOrder.id}-${currentRow.id}`)
+                    ).getByTestId(`select`)
+                ).toHaveTextContent(currentRow.quantityGranted.toString());
+            });
+        });
+    });
+
+    test("Check all button works correctly", () => {
```

Comment:
I'm having two main issues, first i don't know how to actually interact with the checkbox as nothing i do seems to change anything, also I am not too sure how to check whether it is checked, using .toBeChecked() was not working properly and so I looked at other code and they look for a tag called "Mui-checked", which I am not sure is the best way to do this.

Reply:
Yes, this is a tricky one, the key is to add `querySelector()` to get the actual checkbox element. You also need to use an act() block because it's Formik. I only tested this for 1 table since all the tables work the same, but if you want to test all of them you should use a regular for loop instead of .forEach() because the act() calls don't work well with .forEach functions.

You also need a test to test that unchecking works.

```suggestion
    test("Check all button works correctly", () => {
                const { getByTestId } = render(<TeamPendingOrderTable />);
        const currentOrder = mockPendingOrdersInTable[0]
        const checkallBox = getByTestId(`checkall-${currentOrder.id}`).querySelector('input[type="checkbox"]');

        if (checkallBox) {
            await act(async () => {
                fireEvent.click(checkallBox);
                await waitFor(() => {
                    expect(checkallBox).toBeChecked()
                    currentOrder.hardwareInTableRow.forEach((currentRow) => {
                        expect(
                            within(getByTestId(`table-${currentOrder.id}-${currentRow.id}`))
                                .getByTestId(`${currentRow.id}-checkbox`)
                                .querySelector('input[type="checkbox"]')
                        ).toBeChecked();
                    });
                });
            })
        }
    }
```